### PR TITLE
Dw 4896 log adg status to pipeline metadata db

### PR DESCRIPTION
**Pull request by sandeepg-github**
DW-4896 added logic to update dynamodb audit table for success and failure scenarios with correlation_id and run_id for ADG

### CODE REVIEW

File: steps/generate_dataset_from_htme.py [hunk 1]
```diff
@@ -155,15 +174,20 @@ def consolidate_rdd_per_collection(collection, secrets_collections, s3_client,
...
+                        args.correlation_id)
         end_time = time.perf_counter()
         total_time = round(end_time - start_time)
         add_metric("processing_times.csv", collection_name, str(total_time))
-        the_logger.info("Completed Processing of collection : %s for correlation id : %s", collection_name, args.correlation_id)
+        the_logger.info("Completed Processing of collection : %s for correlation id : %s", collection_name,
+                        args.correlation_id)
     except BaseException as ex:
-        the_logger.error(f"Error processing collection for correlation id: %s for collection %s %s", args.correlation_id, collection_name, str(ex))
+        the_logger.error(f"Error processing collection for correlation id: %s for collection %s %s",
```
Comment (UdayChokkam) [inline steps/generate_dataset_from_htme.py:188]: F string can be removed here as the logger can replace the variables

File: steps/generate_dataset_from_htme.py [hunk 2]
```diff
@@ -335,6 +362,51 @@ def get_spark_session():
...
+
+
+def log_start_of_batch(correlation_id, dynamodb=None):
+    """Logging start of batch in metadata audit table as In Progress"""
+    if not dynamodb:
+        dynamodb = get_resource('dynamodb')
+    data_pipeline_audit_table = "${data_pipeline_audit_table}"
+    table = dynamodb.Table(data_pipeline_audit_table)
+    run_id = 1
+    response = table.query(
```
Comment (UdayChokkam) [inline steps/generate_dataset_from_htme.py:376]: Can we deal with exceptions here as there can be connectivity issues sometimes while querying/inserting/updating records in dynamo db

File: steps/generate_dataset_from_htme.py (hunk 1 above)
Comment (sandeepg-github) [reply steps/generate_dataset_from_htme.py:188]: I think this was your change I haven't changed this line of code I have only reformatted

**Review by UdayChokkam — COMMENTED** [review]
I think this approach will leave us with redundant code in adg , pdm and other batch clusters which post status to dynamo db . I would suggest may be move all code to a lambda and call it from all clusters?

**Review by UdayChokkam — APPROVED** [review]

**Pull request merged**